go/vt/schemadiff/table_test.go

```diff
@@ -658,6 +666,37 @@ func TestCreateTableDiff(t *testing.T) {
 			from: "create table t1 (id int primary key, i int, constraint f foreign key (i) references parent(id) on delete cascade)",
 			to:   "create table t2 (id int primary key, i int, constraint f foreign key (i) references parent(id) on delete cascade)",
 		},
+		{
+			name:  "two identical foreign keys, dropping one",
+			from:  "create table t1 (id int primary key, i int, key i_idex (i), constraint f1 foreign key (i) references parent(id), constraint f2 foreign key (i) references parent(id))",
+			to:    "create table t2 (id int primary key, i int, key i_idex (i), constraint f1 foreign key (i) references parent(id))",
+			diff:  "alter table t1 drop foreign key f2",
```

Comment:
Assuming we'll always drop the second one that we parse from left to right and this won't be flaky.

Reply:
That's a good point, and yes this is consistent, but let me check _what_ makes it consistent.

Reply:
OK the order is left-to-right (order of definition), and not `map` iteration order. So it is indeed consistent.